src/components/file-handler/FileHandler.jsx

```diff
@@ -127,28 +144,36 @@ const FileHandler = ({
           rel: 'noopener noreferrer',
         };
 
-  const thumbnail =
-    thumbnailSrc !== undefined ? (
-      <img
-        {...clickProps}
-        src={thumbnailSrc}
-        alt=""
-        className="cursor-pointer"
-      />
-    ) : (
-      <a {...clickProps}>
+  const ThumbnailTag = clickProps.onClick ? 'button' : 'a';
+
+  const thumbnail = (
+    <ThumbnailTag {...clickProps} aria-hidden>
+      {thumbnailSrc !== undefined ? (
+        <img src={thumbnailSrc} alt="" className="cursor-pointer" />
+      ) : (
         <Icon type={iconType} size={24} color={ICON_COLOR['icon-black']} />
-      </a>
-    );
+      )}
+    </ThumbnailTag>
+  );
 
   return (
     <div {...props} className={fileHandlerClass}>
       <div className="sg-file-handler__icon">
         {loading ? <Spinner size="xsmall" /> : thumbnail}
       </div>
       <span className="sg-file-handler__text" ref={textRef}>
+        <span className="sg-visually-hidden" aria-live="polite">
+          {loading
+            ? statusLabel?.loading || 'loading'
+            : statusLabel?.uploaded || 'uploaded'}
+        </span>
         {src !== undefined ? (
-          <Link {...clickProps} size="small" color="text-black">
+          <Link
```

Comment:
@coderitual @domidomi @clxandstuff
Here is that troublesome tag, which is some situations (`onClick` provided`) should be a button

Reply:
`role="button"` added, I followed guidelines from this article: https://www.deque.com/blog/accessible-aria-buttons/

Reply:
thanks for the update @katarzynatobis